src/KingSystem/Sound/sndMusicInfo.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <agl/Utils/aglParameter.h>
+#include <agl/Utils/aglParameterIO.h>
+#include <agl/Utils/aglParameterList.h>
+#include <agl/Utils/aglParameterObj.h>
+#include <container/seadBuffer.h>
+#include <resource/seadResource.h>
+#include "KingSystem/Utils/Types.h"
+
+namespace ksys::snd {
+
+class MusicInfo : public agl::utl::IParameterIO {
+public:
+    template <typename T>
+    struct MinMax : public agl::utl::IParameterObj {
+        inline MinMax(T min, T max, const sead::SafeString& name, const sead::SafeString& label,
+                      const sead::SafeString& meta, agl::utl::IParameterList* parent)
+            : mLabel(label), mMeta(meta), mMin(min, "Min", "最小", mMeta, this),
+              mMax(max, "Max", "最大", mMeta, this) {
+            parent->addObj(this, name);
+        }
+
+        sead::FixedSafeString<64> mLabel;
+        sead::FixedSafeString<32> mMeta;
+        agl::utl::Parameter<T> mMin;
+        agl::utl::Parameter<T> mMax;
+    };
+
+    MusicInfo() : IParameterIO("mscinfo", 0) {}
+
+protected:
+    sead::SafeString mBgmTypeName;
+};
+
+class MusicParam : public MusicInfo {
+public:
+    struct OtherSub : public agl::utl::IParameterObj {
+        OtherSub();
+
+        agl::utl::Parameter<unsigned int> mBPM;
+        agl::utl::Parameter<unsigned int> mPart0FirstWaitCount;
+        agl::utl::Parameter<unsigned int> mPart1FirstWaitCount;
+        agl::utl::Parameter<unsigned int> mPart2FirstWaitCount;
+        agl::utl::Parameter<unsigned int> mPart3FirstWaitCount;
+    };
+
+    MusicParam();
+
+private:
+    agl::utl::ParameterList mHeader;
```

Comment:
```suggestion
    agl::utl::IParameterList mHeader;
```

you can tell this is an IParameterList and not a ParameterList because the IParameterList ctor is called BUT the vtable isn't set to agl::utl::ParameterList